src/puya/index.ts

```diff
@@ -25,10 +25,13 @@ export async function invokePuya({
   options: CompileOptions
   compilationSet: CompilationSet
 }) {
-  ensurePuyaExists()
-  if (!options.skipVersionCheck) {
-    await checkPuyaVersion()
+  let puyaPath = options.puyaPath
+  if (puyaPath) {
+    checkPuyaVersion(puyaPath)
+  } else {
+    puyaPath = await resolvePuyaPath()
```

Comment:
```suggestion
  if (options.puyaPath) {
    checkPuyaVersion(options.puyaPath)
  }
  const puyaPath = options.puyaPath ?? await resolvePuyaPath()
```

Better to declare things as a const if possible.